test/run_test.py

```diff
@@ -155,28 +155,39 @@ def test_cpp_extensions(executable, test_module, test_directory, options):
         print(CPP_EXTENSIONS_ERROR)
         return 1
     cpp_extensions_test_dir = os.path.join(test_directory, 'cpp_extensions')
+    cpp_extensions_with_ninja_dir = os.path.join(cpp_extensions_test_dir, 'ninja_test')
+
+    # Build cpp extension modules that get tested in test_cpp_extensions.py
     return_code = shell([sys.executable, 'setup.py', 'install', '--root', './install'],
                         cwd=cpp_extensions_test_dir)
     if return_code != 0:
         return return_code
+    return_code = shell([sys.executable, 'setup.py', 'install', '--root', './install'],
+                        cwd=cpp_extensions_with_ninja_dir)
+    if return_code != 0:
+        return return_code
```

Comment:
So if ninja is not installed what happens here? Also, for developers who are not testing the vagaries of the build system, it seems suboptimal to force them to sit through two compilations of cpp extensions before they actually get test signal. Better to just have these be completely separate tests, in the same spirit as the distributed tests.